src/components/checkbox/index.js

```diff
@@ -50,37 +69,46 @@ class Checkbox extends BaseComponent {
     /**
      * The selected icon color
      */
-    iconColor: PropTypes.string
+    iconColor: PropTypes.string,
   };
 
   generateStyles() {
     this.styles = createStyles(this.getThemeProps());
   }
 
+  check(value) {
```

Comment:
let's give it a better name so we'll know we're animating something here. 
`toggleCheckAnimation`